include/ncengine/ecs/Component.h

```diff
@@ -87,12 +87,22 @@ struct StoragePolicy : DefaultStoragePolicy {};
 template<PooledComponent T>
 struct ComponentHandler
 {
+    /** @brief Function type for the Factory handler. */
+    using Factory_t = std::function<T(Entity, void*)>;
+
     /** @brief Function type for the DrawUI handler. */
     using DrawUI_t = std::function<void(T&)>;
 
     /** @brief A name for the component with no uniqueness constraints. */
     std::string name = "User Component";
 
+    /** @brief Optional pointer to user data, which is passed to the factory handler. */
+    void* userData = nullptr;
+
+    /** @brief Handler for creating an instance of T.
+     *  @note Factory handlers enable adding/removing types through the editor. */
+    Factory_t factory = nullptr;
```

Comment:
We might be able to conditionally use nullptr or a default lambda factory, depending on whether the type is default constructible.

Reply:
Leaving this out for now. Can adjust later, if needed.